src/OpenCensus.Collector.Dependencies/DependenciesCollector.cs

```diff
@@ -40,10 +41,23 @@ public class DependenciesCollector : IDisposable
         public DependenciesCollector(DependenciesCollectorOptions options, ITracer tracer, ISampler sampler, IPropagationComponent propagationComponent)
         {
             this.diagnosticSourceSubscriber = new DiagnosticSourceSubscriber(
-                new Dictionary<string, Func<ITracer, ISampler, ListenerHandler>>()
+                new Dictionary<string, Func<ITracer, Func<HttpRequestMessage, ISampler>, ListenerHandler>>()
                 { { "HttpHandlerDiagnosticListener", (t, s) => new HttpHandlerDiagnosticListener(t, s, propagationComponent) } },
                 tracer,
-                sampler);
+                x =>
+                {
+                    ISampler s = null;
+                    try
+                    {
+                        s = options.CustomSampler(x);
+                    }
+                    catch (Exception)
+                    {
+                        s = null;
```

Comment:
please add `TODO:` to log error. (Error logging can be implemented as separate PR)

Reply:
Added TODO